packages/grafana-ui/src/components/Select/SelectBase.tsx

```diff
@@ -268,12 +275,41 @@ export function SelectBase<T, Rest = {}>({
   };
 
   if (allowCustomValue) {
-    ReactSelectComponent = Creatable as any;
+    ReactSelectComponent = Creatable;
     creatableProps.allowCreateWhileLoading = allowCreateWhileLoading;
     creatableProps.formatCreateLabel = formatCreateLabel ?? defaultFormatCreateLabel;
     creatableProps.onCreateOption = onCreateOption;
     creatableProps.createOptionPosition = createOptionPosition;
     creatableProps.isValidNewOption = isValidNewOption;
+
+    // code needed to allow editing a custom value once entered
+    // we only want to do this for single selects, not multi
+    if (!isMulti) {
+      creatableProps.inputValue = internalInputValue;
+      creatableProps.onMenuOpen = () => {
+        // make sure we call the base onMenuOpen if it exists
+        commonSelectProps.onMenuOpen?.();
+
+        // restore the input state to the selected value
+        setInternalInputValue(selectedValue?.[0]?.label ?? '');
+      };
+      creatableProps.onInputChange = (val, actionMeta) => {
+        // make sure we call the base onInputChange
+        commonSelectProps.onInputChange(val, actionMeta);
```

Comment:
@ashharrison90
 
you should return what onInputChange returns here as that is important (controls some behaviors)